src/coreclr/System.Private.CoreLib/src/System/Reflection/MdImport.cs

```diff
@@ -194,7 +194,7 @@ internal unsafe struct MetadataEnumResult
         }
     }
 
-    internal readonly struct MetadataImport
+    internal readonly struct MetadataImport : IEquatable<MetadataImport>
```

Comment:
Is it really worth it to add the implementation on internal types that do not actually need?

Note that the linker won't be able to remove the unreachable code since it has to conservatively assume that `IEquatable<T>` is used via reflection.

Reply:
I figured I'd start from the position of "yes" and we can undo them if not desirable.  MetadataImport.Equals is used with a boxed MetadataImport, but only in a Debug.Assert.

Reply:
> Note that the linker won't be able to remove the unreachable code since it has to conservatively assume that IEquatable<T> is used via reflection.

In this case of MetadataImport, the existing private Equals method is called by the Equals override, so presumably the linker can't remove the method anyway.